### PR TITLE
BP-2865 - Make 'Create invoice after shipment' a general setting

### PR DESCRIPTION
**Pull request by LucianTuriacArnia**

### CODE REVIEW

File: Helper/Data.php [hunk 1]
```diff
@@ -254,6 +254,9 @@ public function getStatusCodes(): array
     public function getTransactionAdditionalInfo(array $array, array $rawInfo = [], string $keyPrefix = ''): array
     {
         foreach ($array as $key => $value) {
+            if (in_array($key, ['brq_websitekey', 'brq_signature', 'brq_payer_hash'])) {
+                continue;
+            }
```
Comment (harli91) [inline Helper/Data.php:259]: what do these things do ?

File: Model/Push/DefaultProcessor.php [hunk 2]
```diff
@@ -720,6 +720,13 @@ public function processSucceededPush(string $newStatus, string $message): bool
 
         $this->processSucceededPushAuthorization();
 
+        if ($this->configAccount->getInvoiceHandling() == InvoiceHandlingOptions::SHIPMENT) {
+            $paymentDetails['state'] = Order::STATE_NEW;
+            $paymentDetails['newStatus'] = $this->configAccount->getOrderStatusPending();
```
Comment (harli91) [inline Model/Push/DefaultProcessor.php:725]: why do you use array here, since you send the individual values to the orderRequestService?

Comment (LucianTuriacArnia) [reply Model/Push/DefaultProcessor.php:725]: I used that array here app/code/Buckaroo/Magento2/Model/Push/DefaultProcessor.php:711 
$saveInvoice = $this->invoiceShouldBeSaved($paymentDetails);

$paymentDetails can be changed in invoiceShouldBeSaved

File: Helper/Data.php (hunk 1 above)
Comment (LucianTuriacArnia) [reply Helper/Data.php:259]: This data shouldn't be displayed on Order Transactions and I removed them from Transaction additional information
image

**Pull request merged**